### PR TITLE
FEATURE: Support designating multiple groups as mods on category

### PR DESCRIPTION
**Pull request by OsamaSayegh**
Currently, categories support designating only 1 group as a moderation group on the category. This PR removes the one group limitation and makes it possible to designate multiple groups as mods on a category.

Screenshot:

Internal topic: t/124648.

### CODE REVIEW

File: app/models/reviewable_flagged_post.rb [hunk 1]
```diff
@@ -337,9 +337,13 @@ def unassign_topic(performed_by, post)
...
     user_ids = User.staff.pluck(:id)
 
-    if SiteSetting.enable_category_group_moderation? &&
-         group_id = topic.category&.reviewable_by_group_id.presence
-      user_ids.concat(GroupUser.where(group_id: group_id).pluck(:user_id))
+    if SiteSetting.enable_category_group_moderation? && topic.category
+      user_ids.concat(
+        GroupUser
+          .joins(group: :category_moderation_groups)
+          .where(group: { category_moderation_groups: { category_id: topic.category.id } })
```
Comment (lis2) [inline app/models/reviewable_flagged_post.rb:344]:
```suggestion
          .where(group: { category_moderation_groups.category_id: topic.category.id})
```

File: lib/topic_view.rb [hunk 2]
```diff
@@ -535,16 +535,15 @@ def group_allowed_user_ids
...
+        if SiteSetting.enable_category_group_moderation? && @topic.category.present?
           posts_user_ids = Set.new(@posts.map(&:user_id))
           Set.new(
-            @topic
-              .category
-              .reviewable_by_group
-              .group_users
+            GroupUser
+              .joins(group: :category_moderation_groups)
+              .where(category_moderation_groups: { category_id: @topic.category.id })
```
Comment (lis2) [inline lib/topic_view.rb:543]:
```suggestion
              .where(category_moderation_groups.category_id: @topic.category.id)
```

**Review by lis2 — APPROVED** [review]

File: app/controllers/categories_controller.rb [hunk 3]
```diff
@@ -206,6 +209,10 @@ def update
       old_permissions = cat.permissions_params
       old_permissions = { "everyone" => 1 } if old_permissions.empty?
 
+      if new_group_ids = category_params.delete(:reviewable_by_group_ids).presence
```
[the remark this answers is not in the document]
Comment (OsamaSayegh) [reply app/controllers/categories_controller.rb]: Yeah good catch, thanks! I've fixed it.

File: app/models/category.rb [hunk 4]
```diff
@@ -1113,10 +1113,27 @@ def index_search
     )
   end
 
-  def update_reviewables
-    if should_update_reviewables?
-      Reviewable.where(category_id: id).update_all(reviewable_by_group_id: reviewable_by_group_id)
+  def reviewable_by_group_ids
+    moderating_groups.pluck(:id)
```
[the remark this answers is not in the document]
Comment (OsamaSayegh) [reply app/models/category.rb]: Agreed, done!

**Pull request merged**